lib/pool.js

```diff
@@ -398,6 +439,14 @@ function extend(pool, poolAttrs, poolAlias, oracledb) {
         value: closePromisified,
         enumerable: true,
         writable: true
+      },
+      closing: {
```

Comment:
I guess this new property should be added to `_logStats()` too:
```
  console.log('...closing:', self.closing);
```

Reply:
I agree, that information might be helpful in some cases.
As for the closing, I thought of changing it over to an enum:

*not closing
*close pending
*close issued

Not only it does give a better sense of what is going on under the hood, but also allows execute callbacks to cut in and call close first should there be no connectionsOut in use.

What do you think?

Reply:
Seems fair.